internal/grpc/services/storageprovider/storageprovider.go

```diff
@@ -497,8 +500,11 @@ func (s *service) CreateStorageSpace(ctx context.Context, req *provider.CreateSt
 		}, nil
 	}
 
-	resp.StorageSpace.Id.OpaqueId = storagespace.FormatStorageID(s.conf.MountID, resp.StorageSpace.Id.GetOpaqueId())
-	resp.StorageSpace.Root.StorageId = storagespace.FormatStorageID(s.conf.MountID, resp.StorageSpace.Root.GetStorageId())
+	// The storage driver might set the mount ID by itself, in which case skip this step
+	if mountID, _ := storagespace.SplitStorageID(resp.StorageSpace.Id.GetOpaqueId()); mountID == "" {
```

Comment:
There is no such thing as a _mount id_ in the cs3 api. The Storage Space ID consists of `{provider id}${space id}(!{opaque id})`, which is split into `{storage provider id}` and the `{space id}(!{opaque id})`. We should name this providerID. 

however ... why would the driver know anything about provider ids? When we update the cs3 api we will make the resource id contain three properties: providerid, spaceid any opaqueid. This will get rid of any id wrapping / unwrapping ...but why would you make the driver aware of the provider id? it is just a routing information. ... hmmm the only use case I see for that is when the driver can host multiple providers ... which we might want to support when dealing with migrating spaces between providers ... but for now this seems linke watering the boundaries to me.